buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -342,10 +338,8 @@ static int sizeBucket(int size) {
         return AllocationStatistics.sizeBucket(size);
     }
 
-    @SuppressJava6Requirement(reason = "Guarded by version check")
     @SuppressWarnings("checkstyle:finalclass") // Checkstyle mistakenly believes this class should be final.
-    private static class AllocationStatistics extends StampedLock {
-        private static final long serialVersionUID = -8319929980932269688L;
+    private static class AllocationStatistics {
```

Comment:
@chrisvest we could even just get rid of the whole class and just merge it into `Magazine`. WDYT ?

Reply:
They're separate to organize the code. Maintaining the histograms and the allocation algorithm itself are two complex pieces that don't necessarily need to be in one class, but if it looks fine then I don't mind.

Reply:
We can just keep it as it is ... just wanted to mention it here

Reply:
Also we could just make the `AllocationStatistics` a field in `Magazine` and so not extend it... All of this is purely around coding-style tho and will not change the semantics of the implementation.